src/aiai_eval/named_entity_recognition.py

```diff
@@ -123,8 +124,10 @@ def _check_if_model_is_trained_for_task(self, model_predictions: list) -> bool:
 
         sample_preds = model_predictions[0]
         has_sequence_elements = len(sample_preds[0]) > 0
-        leaves_are_floats = isinstance(sample_preds[0][0], float)
         elements_are_strings = isinstance(sample_preds[0], str)
+        leaves_are_floats = False
+        if not elements_are_strings:
+            leaves_are_floats = sample_preds[0][0].dtype.kind == "f"
```

Comment:
A slight simplification. Mostly for style points, I don't mind if you leave it as is.

```suggestion
        leaves_are_floats = not elements_are_strings and sample_preds[0][0].dtype.kind == "f"
```